### PR TITLE
[docs] fix a few formatting

### PR DESCRIPTION
**Pull request by ydcjeff**
Description: Fix a few docs formatting in `ConfusionMatrix`, neptune logger, engine page and `Paralell` docstring

Some are for fixing docs WARNING when building. There are still 2 WARNINGS left to do which I have no idea how to.

```py
docs/source/utils.rst:11:<autosummary>:1: WARNING: Inline interpreted text or phrase reference start-string without end-string.
docs/source/contrib/handlers.rst:5: WARNING: Could not lex literal_block as "python". Highlighting skipped.
```

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -10,7 +10,7 @@ ignore = E402, E721
 max_line_length = 120
 
 [isort]
-known_third_party=dill,matplotlib,numpy,pkg_resources,pytest,setuptools,sklearn,torch,torchvision,trains
+known_third_party=dill,matplotlib,numpy,pkg_resources,pytest,requests,setuptools,sklearn,torch,torchvision,trains
```
Comment (vfdev-5) [inline setup.cfg:13]: I think this file should not be modified

File: docs/source/engine.rst [hunk 2]
```diff
@@ -133,13 +133,13 @@ Deterministic training
 ``````````````````````
 
 In general, it is rather difficult task to achieve deterministic and reproducible trainings as it relies on multiple
-aspects, e.g. data version, code version, software environment, hardware etc. According to `PyTorch documentation <https://pytorch.org/docs/stable/notes/randomness.html>`_:
+aspects, e.g. data version, code version, software environment, hardware etc. According to `PyTorch documentation <https://pytorch.org/docs/stable/notes/randomness.html>`__:
```
Comment (vfdev-5) [inline docs/source/engine.rst:136]: AFAIK, a link in RST is done as 
```
`name <url>`_
```
For example: https://docutils.sourceforge.io/docs/user/rst/quickref.html#hyperlink-targets

Why do we need to add a dunder ?

File: docs/source/engine.rst [hunk 3]
```diff
@@ -133,13 +133,13 @@ Deterministic training
...
 there are some steps to take in order to make computations deterministic on your specific problem on one specific
 platform and PyTorch release:
 
 - setup random state seed
 
-- set `cudnn to deterministic <https://pytorch.org/docs/stable/notes/randomness.html#cudnn>`_ if applicable
+- set `cudnn to deterministic <https://pytorch.org/docs/stable/notes/randomness.html#cudnn>`__ if applicable
```
Comment (vfdev-5) [inline docs/source/engine.rst:142]: Same question here

File: ignite/distributed/launcher.py [hunk 4]
```diff
@@ -15,7 +15,7 @@ class Parallel:
 
     - XLA on TPUs via `pytorch/xla <https://github.com/pytorch/xla>`_ (if installed)
 
-    - using `Horovod distributed framework <https://horovod.readthedocs.io>`_ (if installed)
+    - using `Horovod distributed framework <https://horovod.readthedocs.io>`__ (if installed)
```
Comment (vfdev-5) [inline ignite/distributed/launcher.py:18]: Here too

**Review by vfdev-5 — COMMENTED** [review]
Thanks for the PR @ydcjeff 
I left few questions/suggestions about the fix you propose.

File: setup.cfg (hunk 1 above)
Comment (ydcjeff) [reply setup.cfg:13]: That's pre-commit that modifies it

File: docs/source/engine.rst (hunk 2 above)
Comment (ydcjeff) [reply docs/source/engine.rst:136]: I found this warning when building docs, found to use [anonymous-hyperlinks](https://docutils.sourceforge.io/docs/ref/rst/restructuredtext.html#anonymous-hyperlinks) to solve from [stackoverflow](https://stackoverflow.com/questions/5464627/how-to-have-same-text-in-two-links-with-restructured-text)
```py
WARNING: duplicate target name: "pytorch documentation"
```

File: docs/source/engine.rst (hunk 3 above)
Comment (ydcjeff) [reply docs/source/engine.rst:142]: Same as above

File: ignite/distributed/launcher.py (hunk 4 above)
Comment (ydcjeff) [reply ignite/distributed/launcher.py:18]: Here too. There is 2 Horovod links (this one and another one below which i didn't edit)

Comment (vfdev-5) [reply ignite/distributed/launcher.py:18]: OK, let's use this one as named hyperlink and below one as anonymous.

File: setup.cfg (hunk 1 above)
Comment (vfdev-5) [reply setup.cfg:13]: I see. I'm not fan of adding it here... but OK.

File: docs/source/distributed.rst [hunk 5]
```diff
@@ -7,7 +7,7 @@ Helper module to use distributed settings for multiple backends:
 
 - XLA on TPUs via `pytorch/xla <https://github.com/pytorch/xla>`_
 
-- using `Horovod distributed framework <https://horovod.readthedocs.io/en/stable/>`_ as a backend
+- using `Horovod framework <https://horovod.readthedocs.io/en/stable/>`_ as a backend
```
Comment (ydcjeff) [inline docs/source/distributed.rst:10]: This has changed to `Horovod framework` as there is `Horovod distributed framework` in `launcher.py`

File: docs/source/engine.rst [hunk 6]
```diff
@@ -133,13 +133,13 @@ Deterministic training
...
 In general, it is rather difficult task to achieve deterministic and reproducible trainings as it relies on multiple
-aspects, e.g. data version, code version, software environment, hardware etc. According to `PyTorch documentation <https://pytorch.org/docs/stable/notes/randomness.html>`_:
+aspects, e.g. data version, code version, software environment, hardware etc. According to `PyTorch note on randomness <https://pytorch.org/docs/stable/notes/randomness.html>`_:
 there are some steps to take in order to make computations deterministic on your specific problem on one specific
 platform and PyTorch release:
 
 - setup random state seed
 
-- set `cudnn to deterministic <https://pytorch.org/docs/stable/notes/randomness.html#cudnn>`_ if applicable
+- set `cudnn to deterministic <https://pytorch.org/docs/stable/notes/randomness.html#cuda-convolution-benchmarking>`_ if applicable
```
Comment (ydcjeff) [inline docs/source/engine.rst:142]: old link didn't exist in stable docs.

File: docs/source/distributed.rst (hunk 5 above)
Comment (vfdev-5) [reply docs/source/distributed.rst:10]: Sounds good !

**Review by vfdev-5 — APPROVED** [review]
Thanks for the PR, Jeff !

**Pull request merged**